File: LeetCode/1/main.py
"""
DIFFICULTY : easy
TAGS : array, hash table
"""


class Solution:
    def twoSum(self, arr: List[int], target: int) -> List[int]:
        mp = {}
        for i in range(len(arr)):
            x = target - arr[i]
            if x in mp:
                return [mp[x], i]
            mp[arr[i]] = i

File: LeetCode/1/main.go
func twoSum(nums []int, target int) []int {
	n := len(nums)
	mp := map[int]int{}

	for i := 0; i < n; i++ {
		val := target - nums[i]
		if ind, exists := mp[val]; exists {
			return []int{ind, i}
		}
		mp[nums[i]] = i
	}

	return []int{}

}